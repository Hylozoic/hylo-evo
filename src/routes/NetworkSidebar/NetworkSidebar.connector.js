import { connect } from 'react-redux'

import getMe from 'store/selectors/getMe'
import getNetworkForCurrentRoute from 'store/selectors/getNetworkForCurrentRoute'

export function mapStateToProps (state, props) {
  const me = getMe(state)
  const network = getNetworkForCurrentRoute(state, props)

  return {
    network: network ? network.ref : null,
    isModerator: network && !!network.isModerator,
    isAdmin: me ? me.isAdmin : false
  }
}

export default connect(mapStateToProps)
