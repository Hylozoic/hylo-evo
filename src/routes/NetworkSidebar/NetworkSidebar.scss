$about-height: 116px;

.network-sidebar {
  padding-top: $space-6x;
}

.settings-link {
  display: block;
  margin-bottom: 20px;
  composes: text-button from 'css/typography.scss';
  &:hover{
    text-decoration: none;
  }
}
