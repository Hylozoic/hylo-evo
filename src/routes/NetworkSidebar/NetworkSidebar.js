import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Link } from 'react-router-dom'

import Loading from 'components/Loading'
import { networkSettingsUrl } from 'util/index'

import './NetworkSidebar.scss'

const { bool, object } = PropTypes

export default class NetworkSidebar extends Component {
  static propTypes = {
    network: object,
    isAdmin: bool
  }

  render () {
    const { isAdmin, network, isModerator } = this.props
    if (!network) return <Loading />

    return <div styleName='network-sidebar'>
      {(isAdmin || isModerator) && <Link
        styleName='settings-link'
        to={networkSettingsUrl(network.slug)}>Settings</Link>}
    </div>
  }
}
