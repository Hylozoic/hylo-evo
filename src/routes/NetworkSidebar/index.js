import NetworkSidebar from './NetworkSidebar'
import connector from './NetworkSidebar.connector'
export default connector(NetworkSidebar)
