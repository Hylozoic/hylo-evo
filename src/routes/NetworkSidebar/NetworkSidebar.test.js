import React from 'react'
import { shallow } from 'enzyme'

import NetworkSidebar from './NetworkSidebar'

const network = {
  id: '1',
  name: 'Networky Network',
  slug: 'networky-network',
  description: 'Blarg blarg'
}

describe('NetworkSidebar', () => {
  it('matches the last snapshot', () => {
    const wrapper = shallow(<NetworkSidebar network={network} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('shows the settings link if a moderator', () => {
    const wrapper = shallow(<NetworkSidebar network={network} isModerator />)
    expect(wrapper).toMatchSnapshot()
  })

  it('shows the settings link if a admin', () => {
    const wrapper = shallow(<NetworkSidebar network={network} isAdmin />)
    expect(wrapper).toMatchSnapshot()
  })
})
