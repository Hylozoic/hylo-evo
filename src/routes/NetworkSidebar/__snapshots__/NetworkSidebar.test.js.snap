// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkSidebar matches the last snapshot 1`] = `
<div
  data-stylename="network-sidebar"
/>
`;

exports[`NetworkSidebar shows the settings link if a admin 1`] = `
<div
  data-stylename="network-sidebar"
>
  <Link
    data-stylename="settings-link"
    replace={false}
    to="/n/networky-network/settings"
  >
    Settings
  </Link>
</div>
`;

exports[`NetworkSidebar shows the settings link if a moderator 1`] = `
<div
  data-stylename="network-sidebar"
>
  <Link
    data-stylename="settings-link"
    replace={false}
    to="/n/networky-network/settings"
  >
    Settings
  </Link>
</div>
`;
